### ISSUE TITLE
Bag Builder (longterm hodl) option & separate section for it

### ISSUE THREAD

**Issue by HomeY** [taniman/profit-trailer-enhancements#355]
Hi,

I think a 'Bag Builder' option would be a really nice enhancement to ProfitTrailer.
The goal of this enhancement would be to keep your profits in the coin you purchased, to build a LongTerm porto (of specific coins). In other words: Once all your sell triggers get hit, you take out your initial investment and keep the profits in the coin you purchased.

Could be enabled/disabled on global or coin level with settings like:
DEFAULT_bagbuilder_enabled = false
COIN_bagbuilder_enabled = true

A separate section (something like 'Pairs (LT)' or 'HODL') could then show these positions, and the value of them should be kept out of your TCV (so it won't mess up initial_cost_percentage or DCA).

**Comment by NOP-Commander** [comment]
Hi @HomeY,

i basically like this idea and was thinking about it too lately but came to a point where i had no clever solution for how ProfitTrailer could handle this in a bullet-proof way.

As far as i know this strategy is used for long-term holding a coin and accumulating more coins with swing-trades. This would imply ProfitTrailer must re-buy those coins with a lower price, compared to the last price when the coin was sold. Any other strategy would not steadily increase the number of coins (the opposite will happen most likely) but "just" create profits with a single coin. Please correct me if I'm wrong.

So from my understanding you need a bullet-proof buy/sell-strategies to ensure accumulation.

Is this possible anyway? Can this e.g. be achieved with golden-/death-cross strategy using 4h candles? Or any other strategy and under any circumstance?

If there is no bullet-proof strategy i will predict a lack of user-satisfaction. Just or mostly creating profits with a single coin without accumulation is (at least for me) not an option.

**Comment by HomeY** [comment]
Quote: ProfitTrailer must re-buy those coins with a lower price, compared to the last price when the coin was sold.

This isn't important, since once the sell gets done and the initial investment is taken out, the rest of the position is free. The only thing that needs to be done then is that the coin value is listed in the (for example) 'HODL' section, the value shouldn't be added to the TCV (since we aren't using it as balance to execute other orders). The entire value of the coins in the 'HODL' section can be seen as profit, since the positions are free, and it should be 'blocked' for selling.
It doesn't matter against what price the orders are done, as long as the initial investment is taken out and there are coins/tokens left that can be added to the HODL position.

Any strategy that results in profit will do, even a simple buy strategy on low STOCH/RSI and a sell strategy on Gain.

**Comment by HomeY** [comment]
Sorry for the late reply (somehow missed the notification).

It could work for 1 coin, with a whitelist/blacklist or ALL coins.
The only difference to how things are working now is that currently, the trader sells the entire position. What it needs to do is keep track of the initial investment (which it already does; Bought Cost), and once the sell triggers get hit, sell the amount of Coin_XXX for the Bought Cost. That way the initial investment (including costs) get sold, and the profits are kept in Coin_XXX.

Example:
You buy 10 ETH, and your gain trigger is set @ 5%. Once the trigger is hit, your position will be 10.5 ETH. The trader then sells 10 ETH and keeps the 0.5 ETH profit (costs left out for the ease of the example). That 0.5 ETH should be registered as profit, and listed on a separate tab and not be used in any other calculation (like TCV). Rinse, wash, repeat ;)

So you see it doesn't matter when and for what price the coin is bought, as long as your buy triggers are hit.

**Comment by ridderedward** [comment]
Was about to make a request for this. 

This option/function would be really nice. To accumelate freecoins with profittrailer. Exactly as in the example. 

You buy X coin for 2 ETH. X coin gains 2% you only sell the 2 ETH amount and keep the 2% profit of X coin. Rince and repeat. So in the end we have more of X Coin.

**Comment by NOP-Commander** [comment]
Not so long ago, the developers of another bot i'm using (won't name it, i don't do promo) introduced an option to "accumulation": Doing "profits" or let's better say avoiding losses. The feature is called "short positions". When you know that the value of an asset will drop, you click on "short position" and technically the following will happen.

1. The position get's sold. You had an amount of X coins.
2. The total selling price Y gets reserved for buying back later.
3. The sold position gets tracked in a separate view where "profit" is indicated while price declines further. This is no real profit but the money that you would have lost if you haven't shorted the position.
4. At some time you close the position manually or by strategy (you can even remove the short-position resulting in reserved funds for this position becoming available again).
5. The bot rebuys the same amount, precisely X coins, for a lower price. The difference between Y and the total selling-price for rebuying is your "profit" which will go back into available funds.

So basically you can avoid losses with this technique too. This is the other option to "accumulation". Both serve a purpose and could increase the UX of ProfitTrailer. The other bot has a fairly different concept - single positions for every buy (incl. split/merge) instead of just one positions per coin - but this shouldn't really matter.

Cheers!

**Comment by ridderedward** [comment]
thank you for your reply man. I do know of some organisation do something that looks like that but I am not sure we are talking about the same company mine has a certain number in it's name. 

Is it possible you share your other bot with me? As I am curious to a solution like this. 

Cheers.

**Comment by NOP-Commander** [comment]
Google for "shorting bot" and you'll find the answer on page 1.

**Comment by taniman** [comment]
Yuo can reserve coins manually. Other than that PT is a trading bot not a hodling bot
